theano/tensor/nnet/abstract_conv.py

```diff
@@ -2311,19 +2421,18 @@ def perform(self, node, inp, out_):
                           for i in range(self.convdim))
 
         if mode == "full":
-            mode = tuple(dil_shape[i] - 1 for i in range(self.convdim))
+            mode = tuple((dil_shape[i] - 1,) * 2 for i in range(self.convdim))
         elif mode == "half":
-            mode = tuple(dil_shape[i] // 2 for i in range(self.convdim))
+            mode = tuple((dil_shape[i] // 2,) * 2 for i in range(self.convdim))
         if isinstance(mode, tuple):
```

Comment:
The check in the `perform` for gradInputs is nicer. This just checks if mode might be equal to `valid`, whereas the check you do in that other `perform` below also checks if `mode` is a tuple of all zeros.